slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -121,8 +117,9 @@ trait MySQLProfile extends JdbcProfile { profile =>
       sym.flatMap(_.findColumnOption[RelationalProfile.ColumnOption.Length]) match {
         case Some(l) => if(l.varying){
             //http://dev.mysql.com/doc/refman/5.7/en/string-type-overview.html
-            if(l.length <= 65535 ) s"VARCHAR(${l.length})" 
-            else if(l.length <= 16777215 ) "MEDIUMTEXT"
+            if(l.length < 65535 ) s"VARCHAR(${l.length})"
+            else if(l.length >= 65535 && l.length < 16777215) "TEXT"
+            else if(l.length <= 16777215) "MEDIUMTEXT"
```

Comment:
Why exclude the 65535 case from VARCHAR? MEDIUMTEXT now only seems to be used for length exactly = 16777215. Was that intended?